application/F3DOptionsParser.cxx

```diff
@@ -402,22 +406,37 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
     this->DeclareOption(grp7, "interaction-test-play", "", "Path to an interaction log file to play interaction events from when loading a file", appOptions.InteractionTestPlayFile, LocalHasDefaultNo, MayHaveConfig::YES,"<file_path>");
     // clang-format on
 
-    cxxOptions.positional_help("file1 file2 ...");
-    cxxOptions.parse_positional({ "input" });
-    cxxOptions.show_positional_help();
     cxxOptions.allow_unrecognised_options();
 
     if (parseCommandLine)
     {
       auto result = cxxOptions.parse(this->Argc, this->Argv);
 
-      auto unmatched = result.unmatched();
+#ifndef F3D_NO_DEPRECATED
+      for (const std::string& input : deprecatedInputs)
+      {
+        /* `deprecatedInputs` may contain an empty string instead of being empty itself */
+        if (!input.empty())
+        {
+          f3d::log::warn("--input option is deprecated, please use positional arguments instead.");
```

Comment:
can it be empty though ?

Reply:
Actually we'd prefer it empty and we want to warn of the deprecation if it is not.
But we cannot do `if(!deprecatedInputs.empty()) warn()` because sometimes (I don't understand why) `deprecatedInputs` comes up as containing an empty string (`{"", }`) instead of empty (`{}`, which also happens, as expected). So we have to check for any non empty element and warn then.

TBH I haven't wrapped my head around why/how the arguments are parsed multiple times and I'm just mitigating the mysterious (to me) side-effects. Hopefully It's not breaking anything along the way.